### PR TITLE
Add support to listen on custom port numbers.

### PR DESCRIPTION
**Pull request by flora-five**
I need to run bind listening on another port than the default and this change would allow me to do that. Please consider merging it, thanks!

### CODE REVIEW

**Comment by GR360RY** [comment]
@flora-five , thanks for the contribution! Allowing to specify listen-on port is a welcome addition.

I think we can simplify the code a bit. See my comments below.

File: defaults/main.yml [hunk 1]
```diff
@@ -22,10 +22,22 @@ bind_dns_keys: []
...
+# the bind_listen_ipv4 list. If this list is shorter than the bind_listen_ipv4
+# list, then the default port is used.
+bind_listen_ipv4_port:
+  - 53
+
 # List of IPv6 address of the network interface(s) to listen on.
 bind_listen_ipv6:
   - "::1"
 
+# The list of ports to listen on for IPv6 addresses, one port for each item in
```
Comment (GR360RY) [inline defaults/main.yml:35]: Replace with: "The list of ports to listen on for IPv6 addresses"

**Review by GR360RY — CHANGES_REQUESTED** [review]
Please take a look at the proposed changes and comment if this will addresses your case.

**Comment by flora-five** [comment]
@GR360RY Thanks for the feedback!

If I understand correctly, the logic you're proposing is the following: listen on all `address:port` combinations by using the product of the `bind_listen_ipv?` and `bind_listen_ipv?_port` lists. Am I right?

The logic in my initial change is to be able to listen on a specific port for every address in the `bind_listen_ipv?` lists. I guess that although my initial logic is more flexible, it may not be necessary in most cases.

Your proposed logic would address my use case and I will proceed then to make the changes.

**Comment by GR360RY** [comment]
@flora-five , thanks for the changes. Last thing - can you please squash changes and force push ?

**Comment by GR360RY** [comment]
@flora-five , squashed commits by myself. Thanks again for the contribution.

**Pull request merged**